metloom/pointdata/files.py

```diff
@@ -0,0 +1,253 @@
+
+    @classmethod
+    def from_station_id(cls, station_id):
+        result = None
+        for e in cls:
+            if station_id == e.station_id:
+                result = e
+                break
+        return result
+
+    @property
+    def point(self):
+        return gpd.points_from_xy([self.longitude],
+                                  [self.latitude],
+                                  [self.elevation])[0]
+
+
+class CSVPointData(PointData):
+    """
+    Some met station data is stored off in flat csv files. This class enables the
+    management of downloading those files while still allowing a similar interface
+    """
+    ALLOWED_VARIABLES = VariableBase
+    ALLOWED_STATIONS = StationInfo
+    UTC_OFFSET_HOURS = 0  # Allows users to specificy the timezone of the datasets
+
+    def __init__(self, station_id, name=None, metadata=None, cache='./cache'):
+        """
+        See docstring for PointData.__init__
+        """
+        super(CSVPointData, self).__init__(
+            station_id, name,
+            metadata=metadata
+        )
+        self._raw_metadata = None
+        self._tzinfo = timezone(timedelta(hours=self.UTC_OFFSET_HOURS))
+        self._cache = Path(cache)
+        self._station_info = None
+        self.valid = False
+
+    def _verify_station(self):
+        """ Verifies the station is valid using the associated enum"""
+        self._station_info = self.ALLOWED_STATIONS.from_station_id(self.id)
+
+        if self._station_info is not None:
+            # Auto assign name
+            if self.name is None:
+                self.name = self._station_info.station_name
+
+            return True
+        else:
+            LOG.error(f"Station ID {self.id} is not valid, allowed id's are "
+                      f"{', '.join(self.ALLOWED_STATIONS.all_station_ids())}")
+            return False
+
+    def _verify_sensor(self, resp_df, variable: SensorDescription):
+        """ Verifies the station is valid using the associated enum"""
+        if variable.code in resp_df.columns:
+            return True
+        else:
+            LOG.debug(f"{variable.name} not found in {self.id} data")
+            return False
+
+    def _file_urls(self, *args):
+        """Returns the url to the file containing the station data"""
+        raise NotImplementedError('CSVPointData._file_urls() must be implemented to '
+                                  'download csv station data.')
+
+    def _assign_datetime(self, resp_df):
+        raise NotImplementedError('CSVPointData._assign_datetime() must be implemented '
+                                  'to download csv station data.')
+
+    def _download(self, urls):
+        """Download the file(s)"""
+        filenames = []
+        for url in urls:
+            with requests.get(url, stream=True) as r:
+                filename = self._cache.joinpath(Path(url).name)
+                filenames.append(filename)
+
+                if not filename.exists():
```

Comment:
Should the `requests.get` move to below this?

Reply:
lol yes. Yes it should be. Fixed.